doc/design/binst-install-command.md

```diff
@@ -0,0 +1,308 @@
+# Design Document: `binst install` Command
+
+## Overview
+
+This document outlines the design for implementing the `binst install` command that provides a native Go implementation of the installation process, achieving script-parity with the generated shell installers.
+
+## Goals
+
+1. **Script Parity**: The `binst install` command must produce identical results to generated installer scripts
+2. **Reusability**: Create modular packages that can be reused for future commands like `binst x`
+3. **Native Implementation**: Avoid shell execution; use Go for all operations
+4. **Cross-Platform**: Support the same platforms as generated installers (Linux, macOS, Windows)
+
+## Non-Goals
+
+1. Windows PowerShell support (this issue)
+2. Alternative sources beyond GitHub (GitLab, S3, etc.)
+3. Additional verification methods beyond current installer capabilities
+4. UX improvements beyond script parity
+
+## Architecture
+
+### Package Structure
+
+```
+pkg/
+├── config/          # Config loading and discovery
+├── resolve/         # Asset resolution logic
```

Comment:
Isn't `pkg/asset.FilenameGenerator` enough? I don't think we need another wrapper package.